src/webui/www/private/scripts/client.js

```diff
@@ -282,8 +303,42 @@ window.addEvent('load', function() {
         return false;
     };
 
+    const removeTorrentFromTagList = function(hash) {
+        if ((hash === null) || (hash === ""))
+            return false;
+
+        let removed = false;
+        for (let key in tagList) {
+            const tag = tagList[key];
+            if (Object.contains(tag.torrents, hash)) {
+                removed = true;
+                tag.torrents.splice(tag.torrents.indexOf(hash), 1);
+            }
+        }
+        return removed;
+    };
+
+    const addTorrentToTagList = function(torrent) {
+        removeTorrentFromTagList(torrent['hash']);
+        if (typeof torrent['tags'] === 'undefined')
+            return false;
+        if (torrent['tags'].length === 0) // No tags
+            return true;
+
+        const tags = torrent['tags'].split(', ');
```

Comment:
Not sure about including the space in the split string. It seems more common to split just on `,` and then trim the tag. But don't feel super strongly about this.